### PR TITLE
[DocsGenerator] Fix link generation for lib type reference

### LINKED ISSUE #9928: `crystal docs` fails if I reference a lib constant from a doc comment
## Details

I seem to have found a way to crash `crystal docs`, using Crystal 0.35.1:

```none
[nix-shell:~/repro]$ cat xyz.cr
lib LibFoo
  BAR = 0
end

# `LibFoo::BAR`
def qua
end

[nix-shell:~/repro]$ crystal docs xyz.cr --project-name=jkl
Unhandled type in `kind`: LibFoo (Exception)
  from ../../../build/source/src/compiler/crystal/tools/doc/type.cr:311:0 in 'kind'
  from ../../../build/source/src/compiler/crystal/tools/doc/type.cr:134:5 in 'path_to'
  from ../../../build/source/src/compiler/crystal/tools/doc/type.cr:366:10 in 'path_to'
  from ../../../build/source/src/compiler/crystal/tools/doc/markdown/doc_renderer.cr:171:5 in 'process_line'
  from ../../../build/source/src/compiler/crystal/tools/doc/markdown/parser.cr:130:5 in 'process_paragraph'
  from ../../../build/source/src/compiler/crystal/tools/doc/markdown/parser.cr:14:11 in 'doc'
  from ../../../build/source/src/compiler/crystal/tools/doc/templates.cr:10:25 in 'to_s'
  from ../../../build/source/src/string.cr:4789:5 in 'generate_types_docs'
  from ../../../build/source/src/file.cr:132:5 in 'docs'
  from ../../../build/source/src/compiler/crystal/command.cr:89:7 in 'run'
  from ../../../build/source/src/compiler/crystal.cr:11:1 in '__crystal_main'
  from ../../../build/source/src/crystal/main.cr:105:5 in 'main'
  from __libc_start_main
  from ../sysdeps/x86_64/start.S:122:0 in '_start'
  from ???
Error: you've found a bug in the Crystal compiler. Please open an issue, including source code that will allow us to reproduce the bug: https://github.com/crystal-lang/crystal/issues
```

If I replace `` `LibFoo::BAR` `` with `asdf`, the docs build properly.

## Environment

I installed Crystal using Nix (nixpkgs commit [cb0908a8](https://github.com/NixOS/nixpkgs/tree/cb0908a810631510674e8bfd8fd40d226c824b96/pkgs/development/compilers/crystal)).

```none
[nix-shell:~/repro]$ crystal --version
Crystal 0.35.1 (2020-11-17)

LLVM: 10.0.1
Default target: x86_64-unknown-linux-gnu

[nix-shell:~/repro]$ which crystal
/nix/store/m0bsb3xqy73v0cmpm9zmv77vxl9dxmxg-crystal-0.35.1/bin/crystal

[nix-shell:~/repro]$ uname -rvmpio
3.10.0-1062.4.1.el7.x86_64 #1 SMP Fri Oct 18 17:15:30 UTC 2019 x86_64 unknown unknown GNU/Linux
```

### PR DESCRIPTION
**Pull request by straight-shoota**
Resolves #9928

### CODE REVIEW

File: spec/compiler/crystal/tools/doc/doc_renderer_spec.cr [hunk 1]
```diff
@@ -275,5 +275,14 @@ describe Doc::Markdown::DocRenderer do
...
       end
     end
+
+    it "does not break when referencing lib type (#9928)" do
+      program = semantic("lib LibFoo; BAR = 0; end", wants_doc: true).program
+      generator = Doc::Generator.new(program, [""])
+
+      # TODO: There should not be a link to LibFoo::Bar in the first place
+      # because LibFoo is undocumented
+      assert_code_link(generator.type(program), "LibFoo::BAR", %(<a href="LibFoo.html#BAR">LibFoo::BAR</a>))
```
Comment (Sija) [inline spec/compiler/crystal/tools/doc/doc_renderer_spec.cr:285]: Fixed with #9819

Comment (bcardiff) [reply spec/compiler/crystal/tools/doc/doc_renderer_spec.cr:285]: @straight-shoota let's merge #9819 before this one. Sounds good?

**Review by bcardiff — APPROVED** [review]

**Comment by straight-shoota** [comment]
So the issue with linking lib isn't actually fixed with #9819 because it only excludes nodoc types and lib is not nodoc.

Simply changing to `must_include?` in the `each_namespace` block doesn't seem to fix it, so I'm not going to bother here. It's a really complex thing and should probably be completely re-done.

This PR is good to be merged as is.

**Comment by straight-shoota** [comment]
Btw. there is actually a check for being in a lib type, it just doesn't seem to work in that scenario.

https://github.com/crystal-lang/crystal/blob/605ec43fcfa7286eae3e28ad212514bda0d3f3e3/src/compiler/crystal/tools/doc/generator.cr#L141

**Pull request merged**